### PR TITLE
Add structure view support

### PR DESCRIPTION
**Pull request by klazuka**
This enables:
1. The Structure tool window (cmd-7)
2. The Structure popup (cmd-F12)
3. ctrl-up/down navigation between structure elements in the main text editor

structure window

# Changes

* registers a class for the `lang.psiStructureViewFactory` extension point
* implementation is in the `org.elmlang.intellijplugin.features.structure` package 
* adds icon files for functions, values and types

# Questions

One thing I was unsure about is where to put the logic to decide whether an `ElmValueDeclarationBase` is a function or a value. It seems like it should live on the Psi element itself, but I don't fully understand how GrammarKit does mixins.

Another problematic thing is that my solution for determining whether an `ElmValueDeclaration` is a value or a function will be incorrect when the function is written in point-free style. We could mitigate this by searching the Psi tree for its type annotation, but that would only work when the user added a type annotation. Solving it for all cases would require that we do the type inference in the plugin (a big job). One option would be to avoid distinguishing between functions and values in the first place--but it seems like a useful distinction when trying to orient yourself in a file.

### CODE REVIEW

File: src/main/resources/org/elmlang/intellijplugin/icons/function-32x32.svg [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="svg2"
+   version="1.1"
+   inkscape:version="0.91 r13725"
+   width="32"
+   height="32"
+   viewBox="0 0 32 32"
+   sodipodi:docname="function-32x32.svg"
+   inkscape:export-filename="/Users/keith/dev/elm-intellij-plugin/src/main/resources/org/elmlang/intellijplugin/icons/function@2x.png"
```
Comment (davecahill) [inline src/main/resources/org/elmlang/intellijplugin/icons/function-32x32.svg]: I'm guessing this has no actual effect, but you probably don't want your local path here?

File: src/main/resources/org/elmlang/intellijplugin/icons/type-32x32.svg [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="svg2"
+   version="1.1"
+   inkscape:version="0.91 r13725"
+   width="32"
+   height="32"
+   viewBox="0 0 32 32"
+   sodipodi:docname="type-32x32.svg"
+   inkscape:export-filename="/Users/keith/dev/elm-intellij-plugin/src/main/resources/org/elmlang/intellijplugin/icons/type@2x.png"
```
Comment (davecahill) [inline src/main/resources/org/elmlang/intellijplugin/icons/type-32x32.svg]: As above, local filepath here.

File: src/main/resources/org/elmlang/intellijplugin/icons/value-32x32.svg [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="svg2"
+   version="1.1"
+   inkscape:version="0.91 r13725"
+   width="32"
+   height="32"
+   viewBox="0 0 32 32"
+   sodipodi:docname="value-32x32.svg"
+   inkscape:export-filename="/Users/keith/dev/elm-intellij-plugin/src/main/resources/org/elmlang/intellijplugin/icons/value.png"
```
Comment (davecahill) [inline src/main/resources/org/elmlang/intellijplugin/icons/value-32x32.svg]: Local filepath.

**Comment by reitzig** [comment]
@durkiewicz I see that this was merged a while ago, but it doesn't seem functional in IDEA yet. Is something broken or has structure view not yet been released?

**Comment by klazuka** [comment]
@reitzig it hasn't been released yet. As best I can tell, Kamil isn't using Elm much these days. I've been working on my own re-write of this plugin. It's not yet usable for daily use, but it's getting close. I hope to have something ready to release in December or January.

**Comment by reitzig** [comment]
@klazuka I may be tempted to have a look, if/when you put in on Github.

**Comment by klazuka** [comment]
@reitzig I've released [my re-write](https://github.com/klazuka/intellij-elm) of the plugin, and it includes support for Structure View. It's available now through the JetBrains plugin repository under the name "Elm". You will have to un-install the old plugin before installing the new one.

**Pull request merged**